core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -19,162 +19,237 @@
+    $scope.threadData = threadDataService.data;
+    $scope.getLabelClass = threadStatusDisplayService.getLabelClass;
+    $scope.getHumanReadableStatus = (
+      threadStatusDisplayService.getHumanReadableStatus);
+    $scope.getLocaleAbbreviatedDatetimeString = (
+      oppiaDatetimeFormatter.getLocaleAbbreviatedDatetimeString);
+
     $scope.activeThread = null;
-    _resetTmpMessageFields();
-  };
-
-  $scope.showCreateThreadModal = function() {
-    $modal.open({
-      templateUrl: 'modals/editorFeedbackCreateThread',
-      backdrop: true,
-      resolve: {},
-      controller: ['$scope', '$modalInstance', function($scope, $modalInstance) {
-        $scope.newThreadSubject = '';
-        $scope.newThreadText = '';
-
-        $scope.create = function(newThreadSubject, newThreadText) {
-          if (!newThreadSubject) {
-            warningsData.addWarning('Please specify a thread subject.');
-            return;
+    $rootScope.loadingMessage = 'Loading';
+    $scope.tmpMessage = {
+      status: null,
+      text: ''
+    };
+
+    var _resetTmpMessageFields = function() {
+      $scope.tmpMessage.status = $scope.activeThread ?
+        $scope.activeThread.status : null;
+      $scope.tmpMessage.text = '';
+    };
+
+    $scope.clearActiveThread = function() {
+      $scope.activeThread = null;
+      _resetTmpMessageFields();
+    };
+
+    $scope.showCreateThreadModal = function() {
+      $modal.open({
+        templateUrl: 'modals/editorFeedbackCreateThread',
+        backdrop: true,
+        resolve: {},
+        controller: ['$scope', '$modalInstance', function(
+          $scope, $modalInstance) {
+          $scope.newThreadSubject = '';
+          $scope.newThreadText = '';
+
+          $scope.create = function(newThreadSubject, newThreadText) {
+            if (!newThreadSubject) {
+              warningsData.addWarning('Please specify a thread subject.');
+              return;
+            }
+            if (!newThreadText) {
+              warningsData.addWarning('Please specify a message.');
+              return;
+            }
+
+            $modalInstance.close({
+              newThreadSubject: newThreadSubject,
+              newThreadText: newThreadText
+            });
+          };
+
+          $scope.cancel = function() {
+            $modalInstance.dismiss('cancel');
+          };
+        }]
+      }).result.then(function(result) {
+        threadDataService.createNewThread(
+          result.newThreadSubject, result.newThreadText, function() {
+            $scope.clearActiveThread();
+          });
+      });
+    };
+
+    var _isSuggestionOpen = function() {
+      return $scope.activeThread.status === 'open';
+    };
+
+    var _isSuggestionCurrent = function() {
+      var activeThread = $scope.activeThread;
+      var suggestionExpVersion = activeThread.suggestion.exploration_version;
+      return explorationData.data.version === suggestionExpVersion;
```

Comment:
(Also, may want to change the name to 'isSuggestionValid'.)

Note that, with the current implementation, only one suggestion may be accepted at a time (and the rest will all go out of date immediately.)